### ISSUE TITLE
Atacworks stuck at "waiting for writter to finish..."

### ISSUE THREAD

**Issue by rochevin** [clara-parabricks/AtacWorks#237]
Hello,

Great paper and great tool and thank you to sharing your docker environment to facilitate the use of your tool,

Maybe i'm doing something wrong, but when i'm trying to use `atacworks denoise`, i'm stuck forever when this sentence shows up :
```
Inference #################### [57500/57611] 
Inference #################### [57550/57611] 
Inference #################### [57600/57611] 
Inference time taken: 4012.039s (Load    3.825s,Prediction 4008.112s)
INFO:2021-04-20 15:29:23,602:AtacWorks-main] Waiting for writer to finish...
```
I'm using it within docker :
`docker run --gpus all --shm-size 2G -it -v ~/DeepG4ToolsComparison/:/DeepG4ToolsComparison claraomics/atacworks /bin/bash`

and my command is : `atacworks denoise --noisybw ATAC_entinostat_mean.bw --genome ../hg19.auto.sizes --out_home denoise --distributed --exp_name ATAC_entinostat_mean`

Thanks,
Vincent

**Comment by NachoDave** [comment]
Hi, I also have the same issue using the docker container.

**Comment by ntadimeti** [comment]
Hi @rochevin thanks for your interest in the tool. Can you please run the default tutorial for atacworks denoise : https://clara-parabricks.github.io/AtacWorks/tutorials/tutorial2.html this will help us understand if the problem is with the setup or with your data. 

Thanks.

**Comment by rochevin** [comment]
hi @ntadimeti thanks a lot for your response !
So I did the tutorial on the same docker container and it worked (with all files created)

```
Inference #################### [57600/57611] 
INFO:2021-05-06 15:40:45,963:AtacWorks-main] Waiting for writer to finish...
Writing the output to bigwig files
Inference time taken: 4156.558s (Load  146.284s,Prediction 4010.163s)
Saving config file to ./denoise_config.yaml...
```
And so i tried to use the exact same command but with my file instead : 
```
nohup atacworks denoise \
    --noisybw ATAC_entinostat_mean.bw \
    --genome ../hg19.auto.sizes \
    --weights_path /tutorial2/models/model.pth.tar \
    --out_home "denoise" \
    --exp_name "ATAC_entinostat_mean" \
    --distributed \
    --num_workers 0 &
```
__Still stuck__ (I had to kill the process) but bedgraph was generated this time and not the BigWig files. So I tried to generate the bigwig myself with 
`bedGraphToBigWig ATAC_entinostat_mean_infer.track.bedGraph ../../../hg19.auto.sizes ATAC_entinostat_mean_infer.track.bw` and `LC_COLLATE=C;sort -k1,1 -k2,2n ATAC_entinostat_mean_infer.track.bedGraph -o ATAC_entinostat_mean_infer.track.bedGraph` __with success__ (same with `ATAC_entinostat_mean_infer.peaks.bedGraph`)
This command was needed in order to generate the bigwig with `bedGraphToBigWig` so maybe it stuck here in the code ? 

Anyway I tried with another file with the same command and it __worked correctly__ so maybe something is wrong with my file.

In summary :
- The first time I did not use the arguments `--num_workers 0` and ` --weights_path /tutorial2/models/model.pth.tar`; what's the default weights used if not precised ? 
- The difference between my two tests files is that `ATAC_entinostat_mean` is __normalized__, but when I tried the first time I also did it with the unnormalized file and It was stuck forever too so I don't know...

Any advice ? 

Thanks,
Vincent
